drakrun/drakrun/test/test_vm.py

```diff
@@ -0,0 +1,269 @@
+import pytest
+
+from drakrun.config import InstallInfo
+from drakrun.vm import VirtualMachine
+from drakrun.util import safe_delete
+from _pytest.monkeypatch import MonkeyPatch
+from common_utils import remove_files, tool_exists
+import tempfile
+import subprocess
+import os
+import re
+import logging
+
+
+@pytest.fixture(scope="session")
+def monkeysession(request):
+    mp = MonkeyPatch()
+    yield mp
+    mp.undo()
+
+
+@pytest.fixture(scope="module")
+def patch(monkeysession):
+    if not tool_exists('xl'):
+        pytest.skip("xen is not found")
+
+    def install_patch():
+        return InstallInfo(
+            vcpus=1,
+            memory=512,
+            storage_backend='qcow2',
+            disk_size='200M',
+            iso_path=None,  # not being required
+            zfs_tank_name=None,
+            lvm_volume_group=None,
+            enable_unattended=None,
+            iso_sha256=None
+        )
+    monkeysession.setattr(InstallInfo, "load", install_patch)
+    monkeysession.setattr(InstallInfo, "try_load", install_patch)
+
+    # being yielded so the the monkeypatch doesn't start cleanup if returned
+    yield monkeysession
+
+
+@pytest.fixture(scope="module")
+def test_vm(patch, config):
+    monkeysession = patch
+
+    @property
+    def vm_name(self):
+        return "test-hvm64-example"
+    monkeysession.setattr(VirtualMachine, "vm_name", vm_name)
```

Comment:
```suggestion
    @property
    def vm_name(self):
        return "test-hvm64-example"
    monkeysession.setattr(VirtualMachine, "vm_name", vm_name)
```

If we now have `_format` in VirtualMachine I think that this can be removed? :thinking:

Reply:
yup, right!